proxy-node-gateway/src/main/java/uk/gov/ida/notification/GatewayApplication.java

```diff
@@ -81,10 +84,15 @@ private void registerExceptionMappers(Environment environment) {
         environment.jersey().register(new VspGenerateAuthnRequestResponseExceptionMapper());
         environment.jersey().register(new TranslatorResponseExceptionMapper());
         environment.jersey().register(new SessionAttributeExceptionMapper());
+        environment.jersey().register(new SessionMissingExceptionMapper());
+        environment.jersey().register(new SessionAlreadyExistsExceptionMapper());
         environment.jersey().register(new GenericExceptionMapper());
     }
 
     private void registerResources(GatewayConfiguration configuration, Environment environment) {
+
+        SessionCacheManager sessionCacheManager = new SessionCacheManager(configuration.getRedisServerURI());
```

Comment:
What happens if it can't connect to the redis server?  Or if the redis url is incorrect?  Does the Gateway not start or do you get some kind of exception?

Reply:
So the service underneath does not try to connect till its been handed to the managed service, which is line 111, which then calls start which will connect to the service. If it fails to connect then it will throw an exception.

I don't know if this is the behaviour we want or we want it to fall over to the in memory redis.

Reply:
A couple of suggestions or things to look out for:
- It should be easy to see from the logs that this is what failed on startup.
- K8 should start the Redis service first?  Or it waits or retries for the Redis service to start?